fdb-extensions/src/main/java/com/apple/foundationdb/async/RTree.java

```diff
@@ -0,0 +1,2568 @@
+                // potentially changed. We resort to re-persisting all slots of the target node.
+                // TODO This can be optimized in the future.
+                //
+                updateNodeSlotsForNodes(transaction, Collections.singletonList(targetNode));
+            } else {
+                Verify.verify(targetNode.getKind() == Kind.LEAF);
+                clearNodeSlot(transaction, targetNode.getId(), deleteSlot);
+            }
+
+            // node is not under-flowing -- indicate that we are done fusing at the current node
+            if (!targetNode.isRoot()) {
+                return CompletableFuture.completedFuture(adjustSlotInParent(targetNode)
+                                                         ? NodeOrAdjust.ADJUST
+                                                         : NodeOrAdjust.NONE);
+            }
+
+            // no fuse and no adjustment
+            return CompletableFuture.completedFuture(NodeOrAdjust.NONE); // no fuse and no adjustment
+        } else {
+            //
+            // Node is under min-capacity -- borrow some children/items from the siblings if possible.
+            //
+            final CompletableFuture<List<Node>> siblings =
+                    fetchSiblings(transaction, targetNode);
+
+            return siblings.thenApply(siblingNodes -> {
+                int numSlots =
+                        Math.toIntExact(siblingNodes
+                                .stream()
+                                .mapToLong(siblingNode -> siblingNode.getSlots().size())
+                                .sum());
+
+                final Node tombstoneNode;
+                final List<Node> newSiblingNodes;
+                if (numSlots == siblingNodes.size() * config.getMinM()) {
+                    if (logger.isTraceEnabled()) {
+                        logger.trace("fusing nodes; node={}, siblings={}",
+                                bytesToHex(targetNode.getId()),
+                                siblingNodes.stream().map(node -> bytesToHex(node.getId())).collect(Collectors.joining(",")));
+                    }
+                    tombstoneNode = siblingNodes.get(siblingNodes.size() - 1);
+                    newSiblingNodes = siblingNodes.subList(0, siblingNodes.size() - 1);
+                } else {
+                    if (logger.isTraceEnabled()) {
+                        logger.trace("handling underflow; node={}, numSlots={}, siblings={}",
+                                bytesToHex(targetNode.getId()),
+                                numSlots,
+                                siblingNodes.stream().map(node -> bytesToHex(node.getId())).collect(Collectors.joining(",")));
+                    }
+                    tombstoneNode = null;
+                    newSiblingNodes = siblingNodes;
+                }
+
+                // temporarily underfill targetNode
+                numSlots --;
```

Comment:
```suggestion
                numSlots--;
```

Reply:
fixed all other occurrences of this type as well